nvflare/app_common/workflows/hierarchical_statistics_controller.py

```diff
@@ -0,0 +1,335 @@
+            self.global_statistics, self.client_statistics, statistic_task, hierarchy_config_json
+        )
+
+        self.log_info(fl_ctx, f"task {self.task_name} statistics_flow for {statistic_task} flow end.")
+
+    def _recursively_round_global_stats(self, global_stats):
+        """Apply given precision to the calculated global statistics.
+
+        Args:
+            global_stats: Global stats.
+
+        Returns:
+            A dict containing global stats with applied precision.
+        """
+        if isinstance(global_stats, dict):
+            for key, value in global_stats.items():
+                if key == StC.GLOBAL or key == StC.LOCAL:
+                    for key, metric in value.items():
+                        if key == StC.STATS_HISTOGRAM:
+                            for ds in metric:
+                                for name, val in metric[ds].items():
+                                    hist: Histogram = metric[ds][name]
+                                    buckets = StatisticsController._apply_histogram_precision(hist.bins, self.precision)
+                                    metric[ds][name] = buckets
+                        else:
+                            for ds in metric:
+                                for name, val in metric[ds].items():
+                                    metric[ds][name] = round(metric[ds][name], self.precision)
+                    continue
+                if isinstance(value, list):
+                    for item in value:
+                        self._recursively_round_global_stats(item)
+        if isinstance(global_stats, list):
+            for item in global_stats:
+                self._recursively_round_global_stats(item)
+
+        return global_stats
+
+    def _combine_all_statistics(self):
+        """Get combined global statistics with precision applied.
+
+        Returns:
+            A dict containing global statistics with precision applied.
+        """
+        result = self.global_statistics
+        return self._recursively_round_global_stats(result)
+
+    def _prepare_inputs(self, statistic_task: str) -> Shareable:
+        """Prepare inputs for the given task.
+
+        Args:
+            statistic_task: Statistics task.
+
+        Returns:
+            A dict containing inputs.
+        """
+        inputs = Shareable()
+        target_statistics: List[StatisticConfig] = StatisticsController._get_target_statistics(
+            self.statistic_configs, StC.ordered_statistics[statistic_task]
+        )
+        for tm in target_statistics:
+            if tm.name == StC.STATS_HISTOGRAM:
+                if StC.STATS_MIN in self.global_statistics[StC.GLOBAL]:
+                    inputs[StC.STATS_MIN] = self.global_statistics[StC.GLOBAL][StC.STATS_MIN]
+                if StC.STATS_MAX in self.global_statistics[StC.GLOBAL]:
+                    inputs[StC.STATS_MAX] = self.global_statistics[StC.GLOBAL][StC.STATS_MAX]
+
+            if tm.name == StC.STATS_VAR:
```

Comment:
Consider changing to "elif".

Reply:
Sounds good, done, thanks for pointing it out.